### ISSUE TITLE
Exporting merged declarations doesn't work.

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#24708]
**Code**

File common.ts
```ts
export interface Identifier {
	type: string;
}
```

File node.ts
```ts
import { Identifier } from "./common";

namespace Identifier {
	export function create(type: string): Identifier {
		return { type };
	}
}

export { Identifier };
```

File use.ts
```ts
import { Identifier } from './node';

let id: Identifer = Identifier.create('abc');
```

**Expected behavior:**

No error annotation on the Identifier type annotation

**Actual behavior:**

The type annotation Indentifier has a red underline `Can not find name Identifier`

Hovering over the Identifer in the import statement shows

<img width="372" alt="capture" src="https://user-images.githubusercontent.com/1931590/41001845-651f38ba-6912-11e8-8fac-4d514aabde27.png">

**Comment by mhegazy** [comment]
It is not clear what merging with an exported symbol really means. does it merge with the source, or does it copy the source properties, merge with that and then export a new type. This never worked really, the fact that we do not flag it as an error is a long standing bug.

**Comment by dbaeumer** [comment]
I would expect it to merge as if both are declared in the same file (which works without problems). Why is it different if the two `Identifier` come from different files?

And as you can see the hover knows about both of them.

**Comment by mhegazy** [comment]
It currently semi-works. merging in general means that the two declarations are the same. and the declaration name points to the sum of both declaration.

In places where that is not possible, the compiler generates an error, for instance `var x; class x {}` since the meaning of the merge is not clear.

In this case, due to a bug in the implementation, the compiler does not raise an error for merging with a local alias. Merging with a local alias is rather unclear, did you mean to add a new property to the original declaration? e.g. would all importers of the original interface see a new property? or was it meant to create a local copy of the declaration, add a property to it, and then export that? 

For the first option (augmenting the original declaration), the language has some clear merging constructs for augmenting the original declaration (see module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation)). 

For the second option, the compiler has implementation assumptions about aliases, that makes the it hard to implement this change, in addition it does not follow the same logic like other merges.
